crates/rolldown/src/types/generator.rs

```diff
@@ -13,14 +13,19 @@ pub struct GenerateContext<'a> {
   pub chunk_graph: &'a ChunkGraph,
   pub plugin_driver: &'a SharedPluginDriver,
   pub warnings: Vec<BuildDiagnostic>,
-  pub module_id_to_codegen_ret: Vec<Option<CodegenReturn>>,
+  pub module_id_to_codegen_ret: Vec<Option<CodegenArtifact>>,
 }
 
 pub struct GenerateOutput {
   pub chunks: Vec<InstantiatedChunk>,
   pub warnings: Vec<BuildDiagnostic>,
 }
 
+pub struct CodegenArtifact {
+  pub code: String,
+  pub map: Option<SourceMap>,
+}
```

Comment:
I made CodegenReturn non-construtable with `#[non_exhaustive]` so a wrapper is needed.